app/controllers/vm_common.rb

```diff
@@ -141,8 +136,8 @@ def show(id = nil)
     @showtype = "config"
     @record = identify_record(id || params[:id], VmOrTemplate)
     if @record.nil?
-      referrer = Rails.application.routes.recognize_path(request.referrer)
-      redirect_to :controller => referrer[:controller], :action => referrer[:action]
+      referrer = Rails.application.routes.recognize_path(request.referer)
```

Comment:
The same `request.referrer` is also used in `app/views/miq_request/_prov_field.html.haml` (498)